### PR TITLE
Preflight

### PR DESCRIPTION
**Pull request by kristoferjoseph**
Documents the preflight pattern

### CODE REVIEW

File: app/docs/md/conventions/preflight.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  export async function get() {
+    return {
+      json: {
+        account: {
+          username: 'thisshouldoverride',
+          id: '39nr34n2'
+        }
+      }
+    }
+  }
```
Comment (colepeters) [inline app/docs/md/conventions/preflight.md:96]: Wondering if the following example might demonstrate overrides a little more practically — it seems a bit odd to override an account on a certain route, but using the account to update the page title on the account route might be a bit more expected?

```suggestion
  export async function get() {
    return {
      json: {
        pageTitle: 'My Account (bobsyouruncle)',
        account: {
          username: 'bobsyouruncle',
          id: '23jk23h23'
        }
      }
    }
  }
```

File: app/docs/md/conventions/preflight.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  ```javascript
+  export default function Preflight ({ req }) {
+    return { /* ...Your data here */ }
+  }
+````
+</doc-code>
+
+
+
+### Example of setting the page title using preflight
```
Comment (colepeters) [inline app/docs/md/conventions/preflight.md:38]:
```suggestion
### Setting the page title using preflight
```

File: app/docs/md/conventions/preflight.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    return titleMap[path] || 'My App Name'
+  }
+````
+</doc-code>
+
+
+
+### Example of overriding default preflight data with an API response
```
Comment (colepeters) [inline app/docs/md/conventions/preflight.md:67]:
```suggestion
### Overriding default preflight data with an API response
```

**Review by colepeters — COMMENTED** [review]
Love it! Left one suggestion re: the override example, and some suggestions to keep example headings consistent w/ examples elsewhere on the docs :)

File: app/docs/md/conventions/preflight.md (hunk 1 above)
Comment (kristoferjoseph) [reply app/docs/md/conventions/preflight.md:96]: No I think this is how it should be shown.
The page title is handled in the preflight differently and I don't want to confuse people.
I needed to show a default that was overidden.
Imagine using preflight to mock out an account so development can progress then overriding it when the actual API is impelmented.

Comment (colepeters) [reply app/docs/md/conventions/preflight.md:96]: Yeah, setting page title differently could be confusing, agreed. I was a bit thrown off by changing the account based on the path but I don't have any other ideas coming to mind! 😄

**Review by colepeters — APPROVED** [review]

**Pull request merged**